src/pages/home/home.ts

```diff
@@ -240,8 +259,43 @@ export class HomePage {
   }
 
   private handleOpenUrl(url: string) {
-    if (!this.incomingDataProvider.redir(url)) {
+    this.incomingDataProvider.redir(url, 'HomePage').then((nextView) => {
+      this.redir(nextView);
+    }).catch(() => {
       this.logger.warn('Unknown URL! : ' + url);
+    })
+  }
+
+  private redir(nextView): void {
```

Comment:
It looks like this code block is duplicated 3 times in this PR, which violates the DRY (Don't Repeat Yourself) principle. A more concise approach would be to define the switch statement on the AppComponent and to fire events from the incomingDataService to the AppComponent. The AppComponent would then perform the navCtrl.push action.

Reply:
Agree. I just update de PR.